resources/helm/dask-gateway/templates/gateway/middleware.yaml

```
@@ -0,0 +1,23 @@
+apiVersion: traefik.containo.us/v1alpha1
+kind: Middleware
+metadata:
+  name: {{ include "dask-gateway.fullname" . | printf "clusters-prefix-%s" | trunc 63 | trimSuffix "-" }}
+  labels:
+    {{- include "dask-gateway.labels" . | nindent 4 }}
+spec:
+  stripPrefixRegex:
+    regex:
+      - '{{ .Values.gateway.prefix | trimSuffix "/" }}/clusters/[a-zA-Z0-9.-]+/'
```

This middleware will not work if someone calls `/clusters/name1`, but only if someone calls `/clusters/name1/`. This is a small difference that I've recently run into issues with for other software.

I'm considering this:

```suggestion
      - '{{ .Values.gateway.prefix | trimSuffix "/" }}/clusters/[a-zA-Z0-9.-]+'
```

And if we should make things more explicit by setting this: https://docs.traefik.io/middlewares/stripprefix/#forceslash